homeassistant/components/google_photos/media_source.py

```diff
@@ -0,0 +1,291 @@
+        """
+        if not item.identifier:
+            # Top level view that lists all accounts.
+            return BrowseMediaSource(
+                domain=DOMAIN,
+                identifier=None,
+                media_class=MediaClass.DIRECTORY,
+                media_content_type=MediaClass.IMAGE,
+                title="Google Photos",
+                can_play=False,
+                can_expand=True,
+                children_media_class=MediaClass.DIRECTORY,
+                children=[
+                    _build_account(entry, PhotosIdentifier(cast(str, entry.unique_id)))
+                    for entry in self._async_config_entries()
+                ],
+            )
+
+        # Determine the configuration entry for this item
+        identifier = parse_identifier(item.identifier)
+        entry = self._async_config_entry(identifier.config_entry_id)
+        client = entry.runtime_data
+
+        if identifier.album_media_id is None:
+            source = _build_account(entry, identifier)
+            source.children = [
+                _build_album(
+                    RECENT_PHOTOS_TITLE,
+                    PhotosIdentifier(
+                        identifier.config_entry_id, album_media_id=RECENT_PHOTOS_ALBUM
+                    ),
+                )
+            ]
+            return source
+
+        # Currently only supports listing a single album of recent photos.
+        if identifier.album_media_id != RECENT_PHOTOS_ALBUM:
+            raise BrowseError(f"Unsupported album: {identifier}")
+
+        # Fetch recent items
+        media_items: list[dict[str, Any]] = []
+        page_token: str | None = None
+        while len(media_items) < MAX_PHOTOS:
+            try:
+                result = await client.list_media_items(
+                    page_size=PAGE_SIZE, page_token=page_token
+                )
+            except GooglePhotosApiError as err:
+                raise BrowseError(f"Error listing media items: {err}") from err
+            media_items.extend(result["mediaItems"])
+            page_token = result.get("nextPageToken")
+            if page_token is None:
+                break
+
+        # Render the grid of media item results
+        source = _build_account(entry, PhotosIdentifier(cast(str, entry.unique_id)))
+        source.children = [
+            _build_media_item(
+                PhotosIdentifier(
+                    identifier.config_entry_id, photo_media_id=media_item["id"]
+                ),
+                media_item,
+            )
+            for media_item in media_items
+        ]
+        return source
+
+    def _async_config_entries(self) -> list[GooglePhotosConfigEntry]:
+        """Return active config entries."""
+        entries = self.hass.config_entries.async_entries(
+            DOMAIN, include_disabled=False, include_ignore=False
+        )
+        return [entry for entry in entries if entry.state == ConfigEntryState.LOADED]
```

Comment:
We have a new helper for this since yesterday or so `hass.config_entries.async_loaded_entries(DOMAIN)` (not sure if it includes disabled and ignored by default)